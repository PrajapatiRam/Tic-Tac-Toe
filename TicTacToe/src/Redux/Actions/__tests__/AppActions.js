import mockStore from 'redux-mock-store';
import {
  savePlayer1Name,
  savePlayer2Name,
  savePlayer1Score,
  savePlayer2Score,
} from '../../Actions/AppActions';

const store = mockStore();

beforeEach(() => {
  store.clearActions();
});

it('savePlayer1Name', async () => {
  await store.dispatch(savePlayer1Name('Test user 1'));
  expect(store.getActions()).toMatchSnapshot();
});

it('savePlayer2Name', async () => {
  await store.dispatch(savePlayer2Name('Test user 2'));
  expect(store.getActions()).toMatchSnapshot();
});

it('savePlayer1Score', async () => {
  await store.dispatch(savePlayer1Score(1));
  expect(store.getActions()).toMatchSnapshot();
});

it('savePlayer2Score', async () => {
  await store.dispatch(savePlayer2Score(2));
  expect(store.getActions()).toMatchSnapshot();
});
