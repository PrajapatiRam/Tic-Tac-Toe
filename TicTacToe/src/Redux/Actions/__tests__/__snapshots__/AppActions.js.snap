// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`savePlayer1Name 1`] = `
Array [
  Object {
    "payload": "Test user 1",
    "type": "SAVE_PLAYER1_ACTION",
  },
]
`;

exports[`savePlayer1Score 1`] = `
Array [
  Object {
    "payload": 1,
    "type": "SAVE_PLAYER1_SCORE_ACTION",
  },
]
`;

exports[`savePlayer2Name 1`] = `
Array [
  Object {
    "payload": "Test user 2",
    "type": "SAVE_PLAYER2_ACTION",
  },
]
`;

exports[`savePlayer2Score 1`] = `
Array [
  Object {
    "payload": 2,
    "type": "SAVE_PLAYER2_SCORE_ACTION",
  },
]
`;
